app/src/main/java/com/andryoga/safebox/data/db/secureDao/UserDetailsDaoSecure.kt

```diff
@@ -3,20 +3,29 @@ package com.andryoga.safebox.data.db.secureDao
 import com.andryoga.safebox.data.db.dao.UserDetailsDao
 import com.andryoga.safebox.data.db.entity.UserDetailsEntity
 import com.andryoga.safebox.security.interfaces.HashingUtils
+import com.andryoga.safebox.security.interfaces.SymmetricKeyUtils
 import javax.inject.Inject
 
 class UserDetailsDaoSecure @Inject constructor(
     private val userDetailsDao: UserDetailsDao,
-    private val hashingUtils: HashingUtils
+    private val hashingUtils: HashingUtils,
+    private val symmetricKeyUtils: SymmetricKeyUtils
 ) : UserDetailsDao {
     override suspend fun insertUserDetailsData(userDetailsEntity: UserDetailsEntity) {
-        userDetailsDao.insertUserDetailsData(hash(userDetailsEntity))
+        var entity = hash(userDetailsEntity)
+        entity = encrypt(entity)
+        userDetailsDao.insertUserDetailsData(entity)
     }
 
     override suspend fun getUserDetails(): UserDetailsEntity {
         return userDetailsDao.getUserDetails()
     }
 
+    override suspend fun getHint(): String {
+        val userDetailsEntity = getUserDetails()
```

Comment:
no need to get all data and then pass hint for decryption
you made a method in dao to get hint... call that instead